sopel/trigger.py

```diff
@@ -106,30 +110,40 @@ class PreTrigger:
     component_regex = re.compile(r'([^!]*)!?([^@]*)@?(.*)')
     intent_regex = re.compile('\x01(\\S+) ?(.*)\x01')
 
-    def __init__(self, own_nick, line, url_schemes=None):
+    def __init__(
+        self,
+        own_nick: tools.Identifier,
+        line: str,
+        url_schemes: Optional[Sequence] = None,
+        identifier_factory: IdentifierFactory = tools.Identifier,
+    ):
+        self.make_identifier = identifier_factory
         line = line.strip('\r\n')
         self.line = line
-        self.urls = tuple()
+        self.urls: Tuple[str, ...] = tuple()
         self.plain = ''
 
         # Break off IRCv3 message tags, if present
-        self.tags = {}
+        self.tags: Dict[str, Optional[str]] = {}
         if line.startswith('@'):
             tagstring, line = line.split(' ', 1)
-            for tag in tagstring[1:].split(';'):
-                tag = tag.split('=', 1)
+            for raw_tag in tagstring[1:].split(';'):
+                tag = raw_tag.split('=', 1)
                 if len(tag) > 1:
                     self.tags[tag[0]] = tag[1]
                 else:
                     self.tags[tag[0]] = None
 
+        # Client time or server time
         self.time = datetime.datetime.utcnow().replace(
             tzinfo=datetime.timezone.utc
         )
         if 'time' in self.tags:
+            # ensure "time" is a string (typecheck)
+            tag_time = self.tags['time'] or ''
```

Comment:
This will catch if `self.tags['time']` is `False` or `0`, maybe, but not if it's some other kind of object that isn't falsy. What am I missing about the utility of this check?

(Plus these changes feel a bit out of place in this patch.)

Reply:
The answer is in the comment above: it ensures that "time" is a string, for type-checking purpose, because `self.tags` has the following type:

    Dict[str, Optional[str]]

It means that any value in `self.tags` can be either a `str` (what we want), or `None` (what we don't want): no integer, no boolean, nothing else but string and `None`. As a result, all I'm making sure here is that I do have a string in the end.

See? Type. It all makes sense.